specification/socks5plus/src/main/scripts/org/kaazing/specification/socks5plus/request/bind/connection.request.bind.with.invalid.uri.request.rpt

```diff
@@ -0,0 +1,69 @@
+connect tcp://localhost:1080
+connected
+
+## Write the request bytes for connecting with no-authentication
+# version: SOCKS v5
+write [0x05]
+
+# number of method: 1
+write [0x01]
+
+# method: no authentication required
+write [0x00]
+
+
+## Read the response, expect success
+# version: SOCKS v5
+read [0x05]
+
+# selected method: no authentication required
+read [0x00]
+
+## Write BIND request
+# version: SOCKS v5
+write [0x05]
+
+# BIND command
+write [0x02]
+
+# RSV field
+write [0x00]
+
+# ATYP with URI
+write [0xFE]
+
+# DST.ADDR with invalid URI
+# URI length
+write [0x00 0x00 0x00 0x09]
```

Comment:
To me this looks like field 4 of a Socks4a message with Kaazing uri address type.  The 0x00 0x00 0x00 0x09 is in fact saying "read a name until the null character is seen".  Probably not what we are expecting in a SOCKS5 message.  This could be a longer-term bug I introduced...?  Or we could say that the ATYP 'FE' needs to be processed like this by fiat.





https://en.m.wikipedia.org/wiki/SOCKS: "A server using protocol SOCKS4A must check the DSTIP in the request packet. If it represents address 0.0.0.x with nonzero x, the server must read in the domain name that the client sends in the packet."




—
Sent from Mailbox

On Mon, Sep 28, 2015 at 10:15 AM, Jitendra Kotamraju
<notifications@github.com> wrote:

>> +## Write BIND request
>> +# version: SOCKS v5
>> +write [0x05]
>> +
>> +# BIND command
>> +write [0x02]
>> +
>> +# RSV field
>> +write [0x00]
>> +
>> +# ATYP with URI
>> +write [0xFE]
>> +
>> +# DST.ADDR with invalid URI
>> +# URI length
>> +write [0x00 0x00 0x00 0x09]
> I see that gateway is sending this. Is this correct ? For variable length address, the first byte gives the length, but here you are using 4 bytes. 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/k3po/k3po/pull/217/files#r40578288